### ISSUE TITLE
Operation IDs result in poor method names in generated code

### ISSUE THREAD

**Issue by dima-v-v** [PavlidisLab/Gemma#302]
Retrieving all datasets`/rest/v2/datasets` is named `all`. Retrieving datasets from a platform is named `datasets`. There's even a `dataset1`.

All these names are used in the generated SDK by swagger codegen.

**Comment by arteymix** [comment]
I have renamed all the method to follow essentially the same name we use in Gemma.R.

There is no change to the URLs resulting from these renaming, only the operation IDs in the specification.

Here's a few examples:

-  getDatasets() will become get_datasets() in Python
- getDatasetSamples() will become get_dataset_samples()
- getGeneGeneCoexpression()

I've also hidden endpoints that we might not want to expose right now in the spec.

I noticed that our annotation endpoints share the same prefix as the new search endpoint. We might have to do something about it to avoid confusion.
